tests/unit/strictdoc/backend/sdoc/test_dsl_passthrough.py

```diff
@@ -22,6 +22,11 @@
 from strictdoc.backend.sdoc.writer import SDWriter
 
 
+@pytest.fixture(autouse=True)
```

Comment:
I normally try to avoid such pytest hacks. Could you explain which problem is this solving and maybe we could find a less hacky way?

Reply:
The file lookup for the bibtex testfiles only worked for the PyCharm Pytest run _or_ the "invoke check" Pytest run.
The pytest root-dirs are somehow different here. 
The fixture changes the current working directory to the directory of the current test module. Maybe we can set autouse=False and annotate/change test_170_grammar_document_bibliography() accordingly.

Other solutions?

Reply:
I have taken a closer look at your branch, and I think, I don't have a final solution but just a few thoughts in the direction:

In the MARKER comment: https://github.com/strictdoc-project/strictdoc/pull/811/files#r1066920663, I see that you are calling a function that further parses a file and you are doing that from a Model-type object which is constructed by TextX. My assumption that the TextX-based objects should be pure data containers manipulated from the outside – they should not access the file system and take on the parsing responsibilities.

This is a non-critical difference but normally I prefer to do a more advanced processing, which goes beyond the parsing performed by TextX, in the `strictdoc/backend/sdoc/processor.py`. Now, if you moved the post-processing parsing code to the processor.py, we have our next problem and that is: the underlying magic access to the current working directory of the program, and I think, it is a good time to make that assumption explicit by providing an optional argument to the SDReader(path_to_root: str = os.getcwd()). Now when that is done, in your unit test you would simply make it explicit by providing an argument to the test file folder, similar to how I am doing it in the other places:

```
PATH_TO_THIS_TEST_FOLDER = os.path.dirname(os.path.abspath(__file__))
```

I hope this is not too vaguely described, please let me know.